scripts/private/_sanity-check-universal.js

```diff
@@ -77,28 +77,41 @@ check if the values provided in the .env file for the Universal Channel Mw and t
   }
 
   if (sanityCheck) {
-    const channelBytes = await ucHandler.connectedChannels(0);
-    const channelId = hre.ethers.decodeBytes32String(channelBytes);
-    const envChannelId = config['sendUniversalPacket'][networkName]['channelId'];
-
-    if (channelId !== envChannelId) {
-      sanityCheck = false;
-      console.log(`
+    let counter = 0;
+    let channelId, envChannelId;
+    let channelBytes;
+    let foundChannel = true;
+    // We don't know how many channels are connected to the Universal Channel Mw, so we loop until we get an error
+    do {
+      // Try to get the channel ID at the index
+      try {
+        channelBytes = await ucHandler.connectedChannels(counter);
+      } catch (error) {
+        // If we get an error, it means we reached the end of the list, do not return, just log the error and set foundChannel to false
+        // console.log(`❌ No channel ID at index: ${counter}`);
+        foundChannel = false;
+      }
+      if (!foundChannel) {
+        channelId = hre.ethers.decodeBytes32String(channelBytes);
+        console.log(`Channel ID in UCH contract: ${channelId}`);
+        envChannelId = config['sendUniversalPacket'][networkName]['channelId'];
+      }
```

Comment:
I want to only execute this code when the error has been caught (which I'm expecting) and the flag has been set to false. So this seems correct to me and has been tested

Reply:
Additional thing to add, is that `channelBytes` is expected to be populated on an earlier iteration and even if that's not the case, the following if statement would capture channelBytes being undefined and log the error message

Reply:
Would it make sense to include this then in the catch block if the desired functionality is to execute this code when the error has been caught

Reply:
@kenobijon what you suggested would've worked but personally I found that to be more confusing. I thought that instead I could clarify by pushing the decoding an logging out of the loop so that it's clearer (hopefully). Let me know if you agree